lambda/shared/src/main/scala/feral/lambda/events/SnsEvent.scala

```diff
@@ -0,0 +1,223 @@
+package feral.lambda
+package events
+
+import cats.syntax.all._
+import io.circe.Decoder
+import io.circe.parser.decode
+import io.circe.scodec._
+import scodec.bits.ByteVector
+
+import java.time.Instant
+import java.util.UUID
+import scala.util.Try
+
+/*
+{
+  "Records": [
+    {
+      "EventVersion": "1.0",
+      "EventSubscriptionArn": "arn:aws:sns:TEST",
+      "EventSource": "aws:sns",
+      "Sns": {
+        "Signature": "TEST",
+        "MessageId": "9f0cfa95-e344-4f93-89ab-5e979503ed1f",
+        "Type": "Notification",
+        "TopicArn": "arn:aws:sns:TEST",
+        "MessageAttributes": {
+          "string": {
+            "Type": "String",
+            "Value": "Testing"
+          },
+          "binary": {
+            "Type": "Binary",
+            "Value": "Testing"
+          },
+          "number": {
+            "Type": "Number",
+            "Value": 1e6
+          },
+          "numberString": {
+            "Type": "Number",
+            "Value": "5"
+          },
+          "array": {
+            "Type": "String.Array",
+            "Value": "[1, \"two\", true, false]"
+          },
+          "unsupported": {
+            "Type": "FancyNewType",
+            "Value": "SpecialValueHere"
+          }
+        },
+        "SignatureVersion": "1",
+        "Timestamp": "2022-04-06T01:02:03.456Z",
+        "SigningCertUrl": "TEST",
+        "Message": "Testing Message",
+        "UnsubscribeUrl": "TEST",
+        "Subject": "Test Message"
+      }
+    }
+  ]
+}
+
+==>
+
+SnsEvent(
```

Comment:
Since you've written it out, maybe we check that it parses to this in the test? :)

Reply:
Done!